lib/web/apiserver_test.go

```diff
@@ -2008,6 +2009,142 @@ func TestCreatePrivilegeToken(t *testing.T) {
 	require.NotEmpty(t, privilegeToken)
 }
 
+func TestCreateTOTPSecrets(t *testing.T) {
+	t.Parallel()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+
+	ctx := context.Background()
+	username := "foo@example.com"
+	pack := proxy.authPack(t, username)
+
+	// Acquire a user token.
+	token, err := types.NewUserToken("some-token-id")
+	require.NoError(t, err)
+	token.SetUser(username)
+	token.SetExpiry(env.clock.Now().Add(5 * time.Minute))
+	_, err = env.server.Auth().Identity.CreateUserToken(ctx, token)
+	require.NoError(t, err)
+
+	endpoint := pack.clt.Endpoint("webapi", "mfa", "token", token.GetName(), "totpsecrets")
+	re, err := pack.clt.PostJSON(ctx, endpoint, url.Values{})
+	require.NoError(t, err)
+
+	var qrcode []byte
+	err = json.Unmarshal(re.Bytes(), &qrcode)
+	require.NoError(t, err)
+	require.NotEmpty(t, qrcode)
+}
+
+func TestAddMFADevice(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+	pack := proxy.authPack(t, "foo@example.com")
+
+	// Enable second factor on.
```

Comment:
```suggestion
	// Enable second factor.
```

Technically you are using "optional", not "on", so better to say less imo.